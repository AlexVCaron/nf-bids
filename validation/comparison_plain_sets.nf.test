nextflow_workflow {

    name "Plugin Comparison: Plain Sets"
    script "./main.nf"
    workflow "main_workflow"

    test("asl001 - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/asl001"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_asl.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("asl002 - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/asl002"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_asl.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }

    test("eeg_cbm - Plugin output") {
        when {
            params {
                bids_dir = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples/eeg_cbm"
                libbids_sh = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/libBIDS.sh/libBIDS.sh"
                config = "/home/local/USHERBROOKE/vala2004/dev/bids2nf/plugins/nf-bids/validation/configs/config_eeg.yaml"
            }
        }
        then {
            assert workflow.success
            assert workflow.out.bids_channel
            def results = workflow.out.bids_channel
            collect()
            def json = groovy.json.JsonOutput.toJson(results)
            def serialized = new groovy.json.JsonSlurper().parseText(json)
            assert snapshot(
                groovy.json.JsonOutput.toJson(serialized)
            ).match()
        }
    }
}
