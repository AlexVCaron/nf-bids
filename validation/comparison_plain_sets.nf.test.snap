{
    "asl001 - Plugin output": {
        "content": [
            "[[[\"sub-Sub103\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"T1w\":{\"json\":\"asl001/sub-Sub103/anat/sub-Sub103_T1w.json\",\"nii\":\"asl001/sub-Sub103/anat/sub-Sub103_T1w.nii.gz\"},\"aslcontext\":{\"tsv\":\"asl001/sub-Sub103/perf/sub-Sub103_aslcontext.tsv\"},\"asl\":{\"json\":\"asl001/sub-Sub103/perf/sub-Sub103_asl.json\",\"nii\":\"asl001/sub-Sub103/perf/sub-Sub103_asl.nii.gz\"}},\"filePaths\":[\"asl001/sub-Sub103/anat/sub-Sub103_T1w.json\",\"asl001/sub-Sub103/anat/sub-Sub103_T1w.nii.gz\",\"asl001/sub-Sub103/perf/sub-Sub103_aslcontext.tsv\",\"asl001/sub-Sub103/perf/sub-Sub103_asl.json\",\"asl001/sub-Sub103/perf/sub-Sub103_asl.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-Sub103\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T22:25:21.08364033"
    },
    "eeg_cbm - Plugin output": {
        "content": [
            "[[[\"sub-cbm001\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm001/eeg/sub-cbm001_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm001\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm002\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm002/eeg/sub-cbm002_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm002\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm003\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm003/eeg/sub-cbm003_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm003\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm004\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm004/eeg/sub-cbm004_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm004\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm005\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm005/eeg/sub-cbm005_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm005\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm006\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm006/eeg/sub-cbm006_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm006\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm007\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm007/eeg/sub-cbm007_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm007\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm008\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm008/eeg/sub-cbm008_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm008\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm009\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm009/eeg/sub-cbm009_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm009\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm010\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm010/eeg/sub-cbm010_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm010\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm011\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm011/eeg/sub-cbm011_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm011\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm012\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm012/eeg/sub-cbm012_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm012\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm013\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm013/eeg/sub-cbm013_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm013\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm014\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm014/eeg/sub-cbm014_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm014\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm015\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm015/eeg/sub-cbm015_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm015\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm016\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm016/eeg/sub-cbm016_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm016\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm017\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm017/eeg/sub-cbm017_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm017\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm018\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm018/eeg/sub-cbm018_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm018\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm019\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm019/eeg/sub-cbm019_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm019\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}],[[\"sub-cbm020\",\"NA\",\"NA\",\"task-protmap\"],{\"data\":{\"channels\":{\"tsv\":\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_channels.tsv\"},\"eeg\":{\"edf\":\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_eeg.edf\",\"json\":\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_eeg.json\"},\"events\":{\"tsv\":\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_events.tsv\"}},\"filePaths\":[\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_channels.tsv\",\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_eeg.edf\",\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_eeg.json\",\"eeg_cbm/sub-cbm020/eeg/sub-cbm020_task-protmap_events.tsv\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-cbm020\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"task-protmap\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T22:25:32.269014537"
    },
    "asl002 - Plugin output": {
        "content": [
            "[[[\"sub-Sub103\",\"NA\",\"NA\",\"NA\"],{\"data\":{\"T1w\":{\"json\":\"asl002/sub-Sub103/anat/sub-Sub103_T1w.json\",\"nii\":\"asl002/sub-Sub103/anat/sub-Sub103_T1w.nii.gz\"},\"aslcontext\":{\"tsv\":\"asl002/sub-Sub103/perf/sub-Sub103_aslcontext.tsv\"},\"asl\":{\"json\":\"asl002/sub-Sub103/perf/sub-Sub103_asl.json\",\"nii\":\"asl002/sub-Sub103/perf/sub-Sub103_asl.nii.gz\"},\"m0scan\":{\"json\":\"asl002/sub-Sub103/perf/sub-Sub103_m0scan.json\",\"nii\":\"asl002/sub-Sub103/perf/sub-Sub103_m0scan.nii.gz\"}},\"filePaths\":[\"asl002/sub-Sub103/anat/sub-Sub103_T1w.json\",\"asl002/sub-Sub103/anat/sub-Sub103_T1w.nii.gz\",\"asl002/sub-Sub103/perf/sub-Sub103_aslcontext.tsv\",\"asl002/sub-Sub103/perf/sub-Sub103_asl.json\",\"asl002/sub-Sub103/perf/sub-Sub103_asl.nii.gz\",\"asl002/sub-Sub103/perf/sub-Sub103_m0scan.json\",\"asl002/sub-Sub103/perf/sub-Sub103_m0scan.nii.gz\"],\"bidsParentDir\":\"/home/local/USHERBROOKE/vala2004/dev/bids2nf/tests/data/bids-examples\",\"subject\":\"sub-Sub103\",\"session\":\"NA\",\"run\":\"NA\",\"task\":\"NA\"}]]"
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.10.0"
        },
        "timestamp": "2025-10-28T22:25:26.124169675"
    }
}